### PR TITLE
Add error when trying to compile picongpu with CUDA 7.5 w/o C++11

### PR DESCRIPTION
**Pull request by Flamefire**
This throws a CMake error when trying to compile with nvcc 7.5 but without C++11 due to the problem as described here: https://github.com/ComputationalRadiationPhysics/picongpu/issues/1290

Discussed with @ax3l

### CODE REVIEW

File: src/picongpu/CMakeLists.txt [hunk 1]
```diff
@@ -392,6 +392,14 @@ if(PIC_ENABLE_INSITU_VOLVIS)
 	set(LIBS ${LIBS} ${ICET_CORE_LIBS} ${ICET_MPI_LIBS})
 endif(PIC_ENABLE_INSITU_VOLVIS)
 
+if( (CUDA_VERSION VERSION_EQUAL 7.5) AND
+    (CMAKE_CXX_STANDARD EQUAL 98) )
+    # Kind of invalid usage of floats in constant expressions:
+    # https://github.com/ComputationalRadiationPhysics/picongpu/issues/1290
+    message(FATAL_ERROR "Please enable C++11 with `-std=c++11` or `-DCMAKE_CXX_STANDARD=11`
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: can you keep the line lengths in this one a bit shorter? :)

File: src/picongpu/CMakeLists.txt [hunk 2]
```diff
@@ -392,6 +392,14 @@ if(PIC_ENABLE_INSITU_VOLVIS)
 	set(LIBS ${LIBS} ${ICET_CORE_LIBS} ${ICET_MPI_LIBS})
 endif(PIC_ENABLE_INSITU_VOLVIS)
 
+if( (CUDA_VERSION VERSION_EQUAL 7.5) AND
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: VERSION_GREATER 7.0

File: src/picongpu/CMakeLists.txt [hunk 3]
```diff
@@ -392,6 +392,14 @@ if(PIC_ENABLE_INSITU_VOLVIS)
...
+    (CMAKE_CXX_STANDARD EQUAL 98) )
+    # Kind of invalid usage of floats in constant expressions:
+    # https://github.com/ComputationalRadiationPhysics/picongpu/issues/1290
+    message(FATAL_ERROR "Please enable C++11 with `-std=c++11` or `-DCMAKE_CXX_STANDARD=11`
+                         when compiling with nvcc 7.5")
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: `with nvcc 7.5 and up`

**Comment by Flamefire** [comment]
@ax3l Anything missing here or can this be merged?

**Comment by ax3l** [comment]
I would move it in the `CUDA` CMake section, since it's a problem with nvcc (or to be more precise: it was only allowed to use those constructs in older nvcc versions in C++98).

**Comment by Flamefire** [comment]
Ok. But I had to move the C++11 enabler too

**Comment by ax3l** [comment]
looks good!

**Pull request merged**